discord/ext/commands/cooldowns.py

```diff
@@ -69,7 +70,11 @@ def get_key(self, msg: Message) -> Any:
         elif self is BucketType.member:
             return (msg.guild and msg.guild.id), msg.author.id
         elif self is BucketType.category:
-            return (msg.channel.category or msg.channel).id  # type: ignore
+            return (
+                msg.channel.category.id
+                if isinstance(msg.channel, discord.abc.GuildChannel)
```

Comment:
Hold on, `category` can still be `None`.

```suggestion
                if isinstance(msg.channel, discord.abc.GuildChannel)
                and channel.category
```

Reply:
oh it should of been **msg**.channel.category